### DISCUSSION TITLE
babel-node seems to not transpile JSX in npm react script with Node version 14

### DISCUSSION THREAD

**Discussion by kari565** [babel/babel#13390] [Q&A]
I'm having a big problem with running my npm script using babel-node. This script generates static error pages from react components. With Node versions 10 and 12 it is running successfully, but when I have updated my node version to 14 babel-node seems to stop transpiling JSX. Now I get this error:
```
<MuiThemeProvider theme={theme}>
     ^

 SyntaxError: Unexpected token '<'
     at Loader.moduleStrategy (internal/modules/esm/translators.js:140:18)
     at async link (internal/modules/esm/module_job.js:42:21)
```
My devDependencies in package.json file:
```
 "devDependencies": {
    "@babel/node": "^7.14.2",
    "@babel/preset-env": "^7.14.2",
    "babel-plugin-styled-components": "^1.12.0",
    "prettier": "2.3.0"
  }
```
and script in package.json:
`"generate-error-pages": "babel-node scripts/generate-error-pages.js"`

And this is my .babelrc file:
```
{
  "presets": ["@babel/preset-env", "@babel/react"],
  "plugins": [["styled-components", { "ssr": true }]]
}
```
I can't make it work with Node version 14 like it worked with version 10 and 12.

**Comment by nicolo-ribaudo** [comment]
Could you create a repository that I can use to debug the problem? 🙏

**Reply by kari565** [reply]
Creating repository helped me find the problem. Thanks for asking me for that. In previous versions of Node I had to add
`"type": "module"` 
in package.json because without it there was an exception caused by using import in JS script and script was runing correctly. Now I observed that I don't need `"type": "module"` anymore. in Node 14 this was the problem of described exception. After deleting this from package.json it is now working also in Node 14. Thaks for help :)

**Reply by nicolo-ribaudo** [reply]
Oh well, good to know!
I'm surprised that `type: "module"` was ignored in Node.js 12, but the reason that `@babel/node` stopped working is that Node.js doesn't have a stable loader API yet for ESM, so `@babel/node` and `@babel/register` don't work with native ECMAScript modules.
